stingray/base.py

```diff
@@ -1402,26 +1403,149 @@ def split_by_gti(self, gti=None, min_points=2):
         if gti is None:
             gti = self.gti
 
-        list_of_tss = []
+        slices = []
 
-        start_bins, stop_bins = gti_border_bins(gti, self.time, self.dt)
-        for i in range(len(start_bins)):
-            start = start_bins[i]
-            stop = stop_bins[i]
-
-            if (stop - start) < min_points:
+        for s in self.stream_from_gti_lists([[g] for g in gti]):
+            if np.size(getattr(s, s.main_array_attr)) < min_points:
                 continue
+            slices.append(s)
+        return slices
+
+    def get_idx_from_time_range(self, start, stop):
+        lower_edge, upper_edge = np.searchsorted(self.time, [start, stop])
+        # Searchsorted will find the first number above stop. We want the last number below stop!
+        return lower_edge, upper_edge - 1
+
+    def stream_from_gti_lists(
+        self, new_gti_lists, root_file_name=None, fmt=DEFAULT_FORMAT, only_attrs=None
+    ):
+        """Split the event list into different files, each with a different GTI.
+
+        Parameters
+        ----------
+        new_gti_lists : list of lists
+            A list of lists of GTIs. Each sublist should contain a list of GTIs
```

Comment:
This is just a comment (maybe an extra naive one). The resulting files could be processed in parallel. In particular cases, e.g. with very long observations cut in several files, the lists of GTIs could be read/processed in parallel.